plugins/database/src/db_plugin.cpp

```diff
@@ -2744,6 +2743,7 @@ irods::error db_reg_data_obj_op(
             errMsg << "no permission to update collection '" << logicalDirName << "'";
             addRErrorMsg( &_ctx.comm()->rError, 0, errMsg.str().c_str() );
         }
+        //_rollback("chlRegDataObj");
```

Comment:
The fact that we don't rollback here may be a bug.

I'm going to capture this in an issue as it doesn't affect the issue this PR is attempting to resolve.

Reply:
Created #7260.